src/module/apps/sidebar/actor-directory.ts

```diff
@@ -244,20 +245,67 @@ class ActorDirectoryPF2e extends fa.sidebar.tabs.ActorDirectory<ActorPF2e<null>>
-            name: "PF2E.Actor.Party.Sidebar.RemoveMember",
-            icon: fa.fields.createFontAwesomeIcon("bus").outerHTML,
-            condition: (li) => !!li.closest("[data-party]") && !li.closest(".folder-header"),
-            callback: (li) => {
-                const actorId = li.dataset.entryId;
-                const partyId = li.closest<HTMLElement>("[data-party]")?.dataset.entryId;
-                const actor = game.actors.get(actorId ?? "");
-                const party = game.actors.get(partyId ?? "");
-                if (actor && party instanceof PartyPF2e) {
-                    party.removeMembers(actor.uuid);
-                }
+        const createTradeArgs = (traderActor: ActorPF2e, selfActor?: ActorPF2e | null): TradeRequestData | null => {
+            const token = canvas.tokens.controlled.length === 1 ? canvas.tokens.controlled[0] : null;
+            selfActor ??= token?.actor ?? game.user.character;
+            if (!selfActor) return null;
+            const owners = game.users.filter(
+                (u) => u.active && !u.isSelf && traderActor.testUserPermission(u, "OWNER"),
+            );
+            const assignee = owners.find((u) => !u.isSelf && u.character === traderActor);
+            const traderUser = assignee ?? owners[0];
+            if (!traderUser) return null;
+            const args = { self: { actor: selfActor }, trader: { user: traderUser, actor: traderActor } };
+            return TradeDialog.canTrade(args) ? args : null;
+        };
+        entries.push(
+            {
+                name: TradeDialog.localize("Request.MenuLabel"),
+                icon: fa.fields.createFontAwesomeIcon("money-bill-transfer", { style: "regular" }).outerHTML,
+                condition: (li) => {
+                    if (foundry.applications.instances.has("trade-dialog")) return false;
+                    const actor = game.actors.get(li.dataset.entryId, { strict: true });
+                    if (canvas.tokens.controlled.some((t) => t.actor === actor)) return false;
+
+                    // For the purpose of showing the menu item, find any eligible actor
+                    const token = canvas.tokens.controlled.length === 1 ? canvas.tokens.controlled[0] : null;
+                    const selfActor =
+                        token?.actor ??
+                        game.user.character ??
+                        game.actors.find(
+                            (a) => a.isOwner && a !== actor && a.isOfType("character", "npc") && a.isAllyOf(actor),
+                        );
+
+                    return !!createTradeArgs(actor, selfActor);
+                },
+                callback: (li) => {
+                    const token = canvas.tokens.controlled.length === 1 ? canvas.tokens.controlled[0] : null;
+                    const assigned = game.user.character?.isOfType("character", "npc") ? game.user.character : null;
+                    const selfActor = token?.actor ?? assigned;
```

Comment:
```suggestion
                    const selfActor = token?.actor ?? game.user.character;
```
Would this work? You already check if its a character or npc in the if statement after.

Reply:
It works because it's checked after, and the controlled token should always be preferred over assigned character.